rules/reports/reports.rules

```diff
@@ -0,0 +1,71 @@
+# -*- mode: Snakemake -*-
+#
+# ReportGeneration rules
+
+import pandas
+from io import StringIO
+
+rule all_reports:
+    input:
+        TARGET_REPORT
+
+def parse_trim_summary(f):
+    for line in f:
+        if line.startswith('Input Read Pairs'):
+            vals = re.findall('\D+\: (\d+)', line)
+            keys = ('input', 'both_kept','fwd_only','rev_only','dropped')
+            return(dict(zip(keys, vals)))
+
+def parse_decontam_json(f):
+    data = json.load(f)
+    return(data['data'])
+
+def summarize_qual_decontam(tfile, jfile):
+    tname = os.path.basename(tfile).split('.out')[0]
+    jname = os.path.basename(jfile).split('_summary.json')[0]
+    if tname != jname:
+        raise ValueError('Unmatched qc and decontam files for: %s' % tname)
+    tf = open(tfile)
+    jf = open(jfile)
+    trim_data = parse_trim_summary(tf)
+    decontam_data = parse_decontam_json(jf)
+    tf.close()
+    jf.close()
+    merged_dict = {**trim_data, ** decontam_data}
+    # todo: use yield for pandas DataFrame
+    return(pandas.DataFrame(merged_dict, index=[tname]))
```

Comment:
For sure we can use base python function to do the work here. Can we mark as an Issue for now, and I will get back to it in the near future. Thanks.